applets/notification_area/status-notifier/sn-dbus-menu-item.c

```diff
@@ -203,9 +204,9 @@ sn_dbus_menu_item_new (GVariant *props)
               cairo_surface_destroy (surface);
             }
 
-          item->item = gtk_image_menu_item_new ();
+          item->item = mate_image_menu_item_new ();
           gtk_image_menu_item_set_always_show_image (GTK_IMAGE_MENU_ITEM (item->item), TRUE);
```

Comment:
If ```mate_image_menu_item_set_image```  is used, there is no need to use ```gtk_image_menu_item_set_always_show_image```again

Reply:
Two issues here: using `gtk_image_menu_item_set_always_show_image` with mate_image_menu_item is (presumably) invalid and throws runtime warnings when the panel is run from terminal.

Also note that we have
` if (GTK_IS_IMAGE_MENU_ITEM (item->item))`
at line 441, I would expect it to always test FALSE

though for some reason I still got indicator menu item icons

Reply:
Looks like i need to change `if (GTK_IS_IMAGE_MENU_ITEM (item->item))` to `if (MATE_IS_IMAGE_MENU_ITEM (item->item))` at line 441 and 450?

Reply:
@zhuyaliang see previous post ^^^^

Reply:
sorry, browser cache wasn't updated.

Reply:
Using `if (MATE_IS_IMAGE_MENU_ITEM (item->item))` with last push.